### PR TITLE
:art: [#1021] Make entire Product card clickable

### PR DESCRIPTION
**Pull request by jiromaykin**
New clean branch for [old issue](https://github.com/maykinmedia/open-inwoner/pull/457).
Templates will need some more simplification, because currently they are not very maintainable.

### CODE REVIEW

**Comment by jiromaykin** [comment]
The famous clickable templates problem... I probably made some terrible choices here; feel free to suggest improvements.

File: src/open_inwoner/components/templates/components/Card/ProductCard.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {% if image %}
+        <div class="card__image-container">
+            <img class="card__image" src="{{ image|thumbnail_url:'card-image' }}" alt="{{ image.default_alt_text }}"/>
+        </div>
+        <div class="card__content">
+                <h2 class="h2"><span class="link link__text">{{ title }}</span></h2>
+            {% else %}
+                <div class="card__content">
+                <h2 class="h2"><span class="link link__text">{{ title }}</span></h2>
+            {% endif %}
```
Comment (Bartvaderkin) [inline src/open_inwoner/components/templates/components/Card/ProductCard.html:14]: The indentation is a bit off here and makes it difficult to see the html/template tag nesting.

File: src/open_inwoner/components/templates/components/Card/RenderCard.html [hunk 2]
```diff
@@ -12,9 +12,11 @@
 
     <div class="card__body{% if direction %} card__body--direction-{{ direction }}{% endif %}{% if grid %} card__body--grid{% endif %}">
         {% if title %}
-            <p class="{% if compact %}h4{% else %}h3{% endif %}">
+            {% if compact %}
```
Comment (Bartvaderkin) [inline src/open_inwoner/components/templates/components/Card/RenderCard.html:15]: You can do `and` in if-tags: `{% if title and compact %}`

File: src/open_inwoner/templates/pages/cases/list.html [hunk 3]
```diff
@@ -15,9 +15,11 @@ <h2 class="h2" id="cases">{{ page_title }} ({{ paginator.count }})</h2>
                 {% render_column start=forloop.counter_0|multiply:4 span=4 %}
                     {% render_card compact=True stretch=True title=case.identificatie %}
                         {% render_list %}
+                            <a href="/accounts/cases/{{ case.uuid }}/status/" class="case-list">
```
Comment (Bartvaderkin) [inline src/open_inwoner/templates/pages/cases/list.html:18]: Don't build urls as strings, use the `{% url %}` tag to reverse it from the router: https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#url

I think it'd be like `{% url 'accounts:case_status' object_id=case.uuid %}`

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]
:+1: for saner tags.

File: src/open_inwoner/components/templates/components/Card/RenderCard.html [hunk 4]
```diff
@@ -12,9 +12,11 @@
...
+            <p class="h4">
```
Comment (vaszig) [inline src/open_inwoner/components/templates/components/Card/RenderCard.html:16]: Indentation here can be fixed as well.

File: src/open_inwoner/components/templates/components/Card/DescriptionCard.html [hunk 5]
```diff
@@ -1,13 +1,8 @@
-{% load card_tags icon_tags link_tags i18n grid_tags thumbnail %}
+{% load card_tags icon_tags link_tags button_tags i18n grid_tags thumbnail %}
```
Comment (vaszig) [inline src/open_inwoner/components/templates/components/Card/DescriptionCard.html:1]: `button_tags` template tag is not used anywhere, can be removed.

**Review by vaszig — CHANGES_REQUESTED** [review]
One remark..don't we want the plan cards on home page to be clickable as well?I know this has to do with the product cards, but I am mentioning this because I saw we have updated plan cards as well (plan-list view).

File: src/open_inwoner/components/templates/components/Card/RenderCard.html (hunk 2 above)
Comment (Bartvaderkin) [reply src/open_inwoner/components/templates/components/Card/RenderCard.html:15]: :hear_no_evil: :see_no_evil:

**Comment by jiromaykin** [comment]
Overview of URL's for testing and checking:
## Overview of cards on frontend:
-Home/Themas theme-cards (RenderCard), with or without image, no description, no arrow:
http://127.0.0.1:8000/themas/
-Product Cards with title, description and arrows, no selected image:
http://127.0.0.1:8000/themas/opvoeding-en-gezin/
-small SUB CategoryCard without description or image, just a title:
http://127.0.0.1:8000/themas/zorg-en-ondersteuning/
-Cards from a Theme hierarchy with lots of arrows linking to products (should only become clickable if there are no links inside):
https://open-inwoner-test.maykin.nl/themas/gezond-blijven/
-larger DescriptionCard (inside RenderCard) with image on the left + description:
http://127.0.0.1:8000/themas/zorg-en-ondersteuning/ (huiselijk geweld card)
-Inloggen/login card, (with tinted cards inside):
http://127.0.0.1:8000/accounts/login/
-Register page, (with tinted cards inside):
http://127.0.0.1:8000/accounts/register/
-after oggign in: Theme cards on Home page:
http://127.0.0.1:8000/
-Samenwerken overview cards (after logging in, normal user), RenderCard with date and arrow
http://127.0.0.1:8000/plans/
-Table-like RenderCard in Mijn Aanvragen/Cases (after logging in, normal user) with date and status and arrow
http://127.0.0.1:8000/accounts/cases/open/
-Location card with text only, no links (scroll to bottom of page)
http://127.0.0.1:8000/themas/opvoeding-en-gezin/products/ditiseenredelijklangetitelvan-een-product-en-levert-problemen-op-in-anchor-menu/

**Comment by jiromaykin** [comment]
> One remark..don't we want the plan cards on home page to be clickable as well?

Good idea - I will make a new PR for the Home page cards, because those are  slightly different and will have the same problem (surrounding anchor linkswill break if there are links inside the content).

**Comment by jiromaykin** [comment]
I've made many of the templates on pages very static now; only RenderCard is still very dynamic - hopefully this makes things bit easier for the future already.

**Review by Bartvaderkin — APPROVED** [review]

**Pull request merged**